include/rapidcheck/state/Commands.hpp

```diff
@@ -49,5 +196,81 @@ void showValue(const std::vector<
   }
 }
 
+template <typename CommandType>
+struct ParallelCommands {
+  using Cmd = CommandType;
+  using Cmds = Commands<CommandType>;
+
+  ParallelCommands() {}
+
+  ParallelCommands(const Cmds &prefix, const Cmds &left, const Cmds &right)
+      : prefix(prefix)
+      , left(left)
+      , right(right) {}
+
+  Cmds prefix;
+  Cmds left;
+  Cmds right;
+};
+
+template <typename Cmd, typename Model, typename Sut>
+void runAllParallel(const ParallelCommands<Cmd> &commands, const Model &state, Sut &sut) {
+
+  // Verify pre-conditions for all possible interleavings
+  detail::isValidSequence(commands, state);
+
+  detail::ParallelExecutionResult<Model, Cmd> result;
+
+  // Run serial commands
+  for (const auto &command : commands.prefix) {
+    auto verifyFunc = command->run(sut);
+    result.prefix.emplace_back(
```

Comment:
Nice catch! I get what you mean, but I think this actually will make the less readable (Did an experiment. If you want I can provide it to you or give you some more details about it). 
The same goal can be achieved by not catching exceptions when executing the prefix in isValidExecution. This way the messages are propageted (which they of course should be as this is valuable information being thrown away). I think this is to be preferred. 

Example output:
```
Using configuration: seed=19423419009701365
Falsifiable after 3 tests and 1 shrink

Parallel command sequence Command<DispenserModel, Dispenser>:
Sequential sequence:
Take
First parallel sequence:
Second parallel sequence:


/Users/tobias/Dropbox/Dev/cpp/ticketdispenser/main.cpp:29:
RC_ASSERT(ticket == (model.ticket + 2))

Expands to:
0 == 1
```